modules/raft/src/main/java/org/apache/ignite/raft/jraft/storage/impl/Logs.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.ignite.raft.jraft.storage.impl;
+
+import java.util.List;
+import org.apache.ignite.raft.jraft.Lifecycle;
+import org.apache.ignite.raft.jraft.entity.LogEntry;
+import org.apache.ignite.raft.jraft.option.LogStorageOptions;
+import org.apache.ignite.raft.jraft.storage.Storage;
+
+/**
+ * Log entry storage (used for internal needs of {@link VolatileLogStorage}).
+ */
+interface Logs extends Lifecycle<LogStorageOptions>, Storage {
+    /**
+     * Get logEntry by index.
+     */
+    LogEntry getEntry(final long index);
+
+    /**
+     * Append entries to log.
+     */
+    void appendEntry(final LogEntry entry);
+
+    /**
+     * Append entries to log, return append success number.
+     */
+    void appendEntries(final List<LogEntry> entries);
+
+    /**
+     * Delete logs from storage's head, [first_log_index, first_index_kept) will be discarded.
+     */
+    void truncatePrefix(final long firstIndexKept);
+
+    /**
+     * Delete uncommitted logs from storage's tail, (last_index_kept, last_log_index] will be discarded.
+     */
+    void truncateSuffix(final long lastIndexKept);
+
+    /**
+     * Drop all the existing logs and reset next log index to |next_log_index|. This function is called after installing
```

Comment:
I'm curious about `next_log_index`, `first_log_index`, `last_index_kept` and so on. Are these variables defined in raft paper?

Reply:
Those are internal details of JRaft implementation, not terms from the raft paper

Reply:
Ok!